drizzle-effect/src/index.ts

```diff
@@ -0,0 +1,520 @@
+              >,
+              undefined | null
+            >,
+            Exclude<
+              PropertySignatureType<InsertColumnPropertySignatures<TTable>[S]>,
+              undefined | null
+            >
+          >
+        : InsertColumnPropertySignatures<TTable>[S];
+    }) => InsertColumnPropertySignatures<TTable>[Col] extends Schema.PropertySignature<
+      any,
+      any,
+      any,
+      any,
+      any,
+      any,
+      any
+    >
+      ? Schema.Schema<
+          Exclude<
+            PropertySignatureEncoded<
+              InsertColumnPropertySignatures<TTable>[Col]
+            >,
+            undefined | null
+          >,
+          any
+        >
+      : Schema.Schema<
+          Exclude<
+            Schema.Schema.Encoded<InsertColumnPropertySignatures<TTable>[Col]>,
+            undefined | null
+          >,
+          any
+        >);
+
+type SelectRefineArg<
+  TTable extends Drizzle.Table,
+  Col extends keyof Columns<TTable>,
+> =
+  | Schema.Schema<any, any, any>
+  | ((s: {
+      [S in keyof InsertColumnPropertySignatures<TTable>]: InsertColumnPropertySignatures<TTable>[S] extends Schema.PropertySignature<
+        any,
+        any,
+        any,
+        any,
+        any,
+        any,
+        any
+      >
+        ? Schema.Schema<
+            Exclude<
+              PropertySignatureEncoded<
+                InsertColumnPropertySignatures<TTable>[S]
+              >,
+              undefined | null
+            >,
+            Exclude<
+              PropertySignatureType<InsertColumnPropertySignatures<TTable>[S]>,
+              undefined | null
+            >
+          >
+        : InsertColumnPropertySignatures<TTable>[S];
+    }) => InsertColumnPropertySignatures<TTable>[Col] extends Schema.PropertySignature<
+      any,
+      any,
+      any,
+      any,
+      any,
+      any,
+      any
+    >
+      ? Schema.Schema<
+          Exclude<
+            PropertySignatureEncoded<
+              InsertColumnPropertySignatures<TTable>[Col]
+            >,
+            undefined | null
+          >,
+          any
+        >
+      : Schema.Schema<
+          Exclude<
+            Schema.Schema.Encoded<InsertColumnPropertySignatures<TTable>[Col]>,
+            undefined | null
+          >,
+          any
+        >);
+
+type InsertRefine<TTable extends Drizzle.Table> = {
+  [K in keyof Columns<TTable>]?: InsertRefineArg<TTable, K>;
+};
+
+type SelectRefine<TTable extends Drizzle.Table> = {
+  [K in keyof Columns<TTable>]?: SelectRefineArg<TTable, K>;
+};
+
+const literalSchema = Schema.Union(
+  Schema.String,
+  Schema.Number,
+  Schema.Boolean,
+  Schema.Null,
+);
+
+type Json =
+  | string
+  | number
+  | boolean
+  | {
+      [key: string]: Json;
+    }
+  | Json[]
+  | readonly Json[]
+  | null;
+
+export const Json = Schema.suspend(
+  (): Schema.Schema<Json> =>
+    Schema.Union(
+      literalSchema,
+      Schema.Array(Json),
+      Schema.Record(Schema.String, Json),
```

Comment:
For the latest version this should be
```suggestion
      Schema.Record({ key: Schema.String, value: Json })
```